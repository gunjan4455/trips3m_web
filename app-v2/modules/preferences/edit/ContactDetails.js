import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';

import { DownArrow } from 'helpers/Icon/Icon';
import ISD_CODES from 'helpers/isdCodes';
import { required, phoneNumber, email } from 'utils/formValidators';
import TextBox from '../components/TextBox';
import Dropdown from '../components/Dropdown';
import Header from './Header';

const ContactDetails = ({ handleSubmit, submitting }) => {
  const PREFERENCE_ISD_CODES = Object.keys(ISD_CODES).map(key => {
    return { label: ISD_CODES[key], value: ISD_CODES[key] };
  });

  return (
    <div className='sbcw'>
      <div className='absolute t0 l0 wfull'><Header /></div>

      <div className='p15'>
        <form onSubmit={handleSubmit}>
          <div className='mb15'>
            <label htmlFor="emailId" className="block m0 f14p fw7 sfc6 mb8">Email ID<em className="pfc2">*</em></label>
            <Field
              name="email"
              component={TextBox}
              label="Email Id"
              placeholder="example@domain.com"
              validate={[required, email]}
            />
          </div>

          <div className='mb15'>
            <label htmlFor='phoneNumber' className='block m0 f14p fw7 sfc6 mb8'>Phone Number <em className='pfc2'>*</em></label>
            <div className='row'>
              <div className='col-xs-4'>
                <div className=" select-con input-box">
                  <Field
                    name="isdCode"
                    options={PREFERENCE_ISD_CODES}
                    component={Dropdown}
                    defaultValue="+91"
                  />
                  <span className='r10 block wauto lauto input-icon absolute-center-v downArrow'><DownArrow /></span>
                </div>
              </div>
              <div className='col-xs-8 pl0'>
                <Field
                  name="phoneNo"
                  component={TextBox}
                  placeholder="Mobile/Contact No."
                  validate={[required, phoneNumber]}
                />
              </div>
            </div>
          </div>

          <div className='fixed b0 p8 wfull sbcw bs6 l0 r0 bt z2'>
            <button type='submit' disabled={submitting} className='btn-filled-pri-large wfull at_confirm'>Confirm</button>
          </div>
        </form>
      </div>
    </div>
  );
};

ContactDetails.propTypes = {
  submitting: PropTypes.bool,
  handleSubmit: PropTypes.func
};

ContactDetails.defaultProps = {};

export default reduxForm({
  form: 'ContactDetailsPreferences' // a unique identifier for this form
})(ContactDetails);
