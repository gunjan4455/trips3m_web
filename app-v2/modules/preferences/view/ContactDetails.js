import React from 'react';
import PropTypes from 'prop-types';

import { ContactIcon } from 'helpers/Icon/Icon';

import { GPreferenceDiv } from './G';
import Modal from 'modules/shared/GenericModal';
import ContactDetailsEditForm from '../edit/ContactDetails';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import EditPreferencesPopup from '../components/EditPreferencesPopup';
import { rescue } from 'helpers/utils';

class ContactDetails extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      closeModal: false
    };
  }

  shouldComponentUpdate(nextProps) {
    if (!this.props.updated && nextProps.updated) {
      this.closeModal();
    }
    return true;
  }

  updateContactInfo = (values) => {
    const params = {
      isd_code: values.isdCode,
      requested_trip: {
        email: values.email,
        phone_no: values.phoneNo
      }
    };
    this.trackSegmentEvent(ORGANISM_CLICKED, 'Contact details', '', 'Confirm');
    this.props.handleSubmit(this.props.currentTripId, params);
  };

  trackSegmentEvent = (event, section, object, cta) => {
    rescue(() => {
      this.props.trackSegment(event, section, object, cta);
    });
  };

  handleClick = () => {
    this.trackSegmentEvent(ORGANISM_CLICKED, 'Edit preferences window', 'contact', 'edit');
    this.setState({ closeModal: false });
  };

  afterModalClose = () => {
    this.trackSegmentEvent(ORGANISM_CLICKED, 'Contact details', '', 'Cancel');
  };

  closeModal = () => {
    this.setState({ closeModal: true });
  };

  render() {
    const { preferences, isConvertedTrip, isCancelledTrip, opDetails } = this.props;

    return (
      <div className='sbcw p15 pb0 at_contactDetails'>
        <div className='flex alignCenter spaceBetween mb15'>
          <h6 className='flex alignCenter f16 fw9 m0 flexFull'>
            <span className='block input-24 mr8'><ContactIcon /></span>
            <span className="sfc6 at_headingContectDetail">Contact Details</span>
          </h6>
          {!isConvertedTrip || isCancelledTrip ? <Modal
              trigger={<span className='sfc1 f14 fw7 at_edit1'
                             onClick={() => this.handleClick()}>Edit</span>}
              fullView={true}
              hasCustomClose={true}
              customCloseTrigger={this.state.closeModal}
              onRequestCloseCustom={this.afterModalClose}
            >
              <ContactDetailsEditForm initialValues={preferences}
                                      onSubmit={this.updateContactInfo} />
            </Modal> :
            <Modal trigger={<span className='sfc1 f14 fw7 at_edit1'>Edit</span>}>
              <EditPreferencesPopup opDetails={opDetails} closeModal={this.closeModal} />
            </Modal>}
        </div>
        <div className='flex flexWrap'>
          <GPreferenceDiv className='flexFull mb15 break-word'>
            <p>Email ID</p>
            <p className='at_email'>{preferences.email}</p>
          </GPreferenceDiv>
          <GPreferenceDiv className='flexFull mb15 break-word'>
            <p>Phone Number</p>
            <p className='at_phone'>{preferences.phoneNo ? `${preferences.isdCode}${preferences.phoneNo}` : '-'}</p>
          </GPreferenceDiv>
        </div>
      </div>
    );
  }
}

ContactDetails.propTypes = {
  currentTripId: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
  preferences: PropTypes.object,
  opDetails: PropTypes.object,
  isConvertedTrip: PropTypes.bool,
  isCancelledTrip: PropTypes.bool,
  updated: PropTypes.bool,
  errors: PropTypes.array,
  handleSubmit: PropTypes.func,
  trackSegment: PropTypes.func
};

ContactDetails.defaultProps = {
  trackSegment: () => {
  }
};

export default ContactDetails;
