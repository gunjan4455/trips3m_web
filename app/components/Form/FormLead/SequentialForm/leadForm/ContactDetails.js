import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Contact } from 'helpers/Icon/Icon';
import { Field } from 'redux-form';
import {
  email,
  requiredWithmessage,
  phoneNumberRequired
} from '../../../../../utils/formValidators';
import {
  renderPhone,
  renderEmail,
  genericCheckbox,
} from '../../FormOneVariation';
import {
  SEQUENTIAL_FORM_FIVE, SEQUENTIAL_FORM_TWO, SEQUENTIAL_FORM_SEVEN, SEQUENTIAL_FORM_THREE,
} from '../../../../../constants/Slider';
import Footer from '../footer';

class ContactDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: props.email || '',
    };
  }

  goBack = () => {
    this.props.callNextSlide({
      submitVariation: '',
      variation: true,
      variationStep: SEQUENTIAL_FORM_FIVE,
      isBackButton: true
    });
  };

  goForward = () => {
    this.props.callNextSlide({
      submitVariation: true,
      variation: true,
      variationStep: SEQUENTIAL_FORM_SEVEN
    });
  };

  render() {
    const { formValues } = this.props;
    const goNextDisabled = formValues.email && formValues.phone_no ? false : true;

    return (
      <div className="form-body pt15">
        <div className="contact-icon-container">
          <Contact />
        </div>
        <div className="p15">
          <div className="pb8">
          <Field
            name="email"
            component={renderEmail}
            placeholder="Email"
            type="email"
            label="Email ID"
            validate={[
              requiredWithmessage('Enter valid email id'),
              email,
            ]}
            changeHandler={e => {
              this.setState({ email: e.target.value });
              this.props.handleEmailChange();
            }}
            emailVal={this.state.email}
          />
          </div>
          <Field
            name="phone_no"
            component={renderPhone}
            placeholder="Phone"
            type="number"
            label="Phone No"
            validate={phoneNumberRequired}
            defaultValue={
              this.props.formValues && this.props.formValues.isd_code
                ? this.props.formValues.isd_code
                : ''
            }
            isFocus={!goNextDisabled}
          />
          <Field
            name="whatsapp_opt_status"
            component={genericCheckbox}
            type="checkbox"
            label="Send trip updates on WhatsApp"
            className="checkbox-common sfcw"
            id="whatsapp_opt_status"
            classname="at_whatsapp_opt_status"
            formValues={this.props.formValues}
          />
        </div>
        <Footer goBack={this.goBack} goForward={this.goForward} goNextDisabled={goNextDisabled}/>
      </div>
    );
  }
}

ContactDetails.propTypes = {
  formStage: PropTypes.number,
  formValues: PropTypes.object,
  location: PropTypes.object,
  handleEmailChange: PropTypes.func,
  callNextSlide: PropTypes.func,
  email: PropTypes.string,
  progressBar: PropTypes.func
};

export default ContactDetails;
