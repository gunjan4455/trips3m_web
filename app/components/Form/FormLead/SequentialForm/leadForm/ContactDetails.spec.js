import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ContactDetails from './ContactDetails';
import {
  SEQUENTIAL_FORM_SEVEN,
  SEQUENTIAL_FORM_THREE,
  SEQUENTIAL_FORM_FIVE,
  SEQUENTIAL_FORM_TWO
} from "../../../../../constants/Slider";

const mockCallNextSlide = jest.fn();
const mockDispatchRxFormAction = jest.fn();

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  email: null,
  callNextSlide: mockCallNextSlide,
  dispatchRxFormAction: mockDispatchRxFormAction,
  location: {
    query: {
      variation: '3'
    }
  },

  formValues: {
    email: null,
    dont_utilize_budget_new: true,
    budget_new: true,
    start_date_noidea: false,
    adult: '2',
    child: '0',
    isd_code: '+91',
    child_age_answers_attributes: [
      {
        user_input: '0',
        customization_id: '7',
        option_id: '29'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '30'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '31'
      }
    ],
    syncErrors: {
      email: 'Enter valid email id',
      phone_no: 'Phone number is required'
    },
    to_loc: 'Kerala',
    start_date: '2019-07-25',
    start_date_flexible: false,
    trip_days: '2',
    hotel_needed_answer_attributes: {
      option_id: '95',
      customization_id: '22'
    },
    hotel_category_answers_attributes: [
      {
        option_id: '7',
        customization_id: '2'
      }
    ],
    budget: '5500..6000',
    infant: '0'
  }

};
const shallowWrapperContactDetails = (props) => {
  return shallow(<ContactDetails {...props} />);
};

describe('lead form module', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS, location: { ...DEFAULT_PROPS.location, query: { ...DEFAULT_PROPS.location.query } } };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperContactDetails(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call goForward properly', () => {
    mockCallNextSlide.mockClear();
    const expected_input = {
      submitVariation: true,
      variation: true,
      variationStep: SEQUENTIAL_FORM_SEVEN
    };
    const wrapper = shallowWrapperContactDetails(props);
    wrapper.instance().goForward();
    expect(mockCallNextSlide).toHaveBeenCalledWith(expected_input);
  });

  it('should call goForward properly when variation in query is 4', () => {
    props.location.query.variation = '4';
    mockCallNextSlide.mockClear();
    const expected_input = {
      submitVariation: true,
      variation: true,
      variationStep: SEQUENTIAL_FORM_SEVEN
    };
    const wrapper = shallowWrapperContactDetails(props);
    wrapper.instance().goForward();
    expect(mockCallNextSlide).toHaveBeenCalledWith(expected_input);
  });

  it('should call goBack properly when variation step is sequential form five', () => {
    mockCallNextSlide.mockClear();
    const expected_input = {
      submitVariation: '',
      variation: true,
      isBackButton: true,
      variationStep: SEQUENTIAL_FORM_FIVE
    };
    const wrapper = shallowWrapperContactDetails(props);
    wrapper.instance().goBack();
    expect(mockCallNextSlide).toHaveBeenCalledWith(expected_input);
  });


  it('should call goBack properly', () => {
    props.location.query.variation = '4';
    mockCallNextSlide.mockClear();
    const expected_input = {
      submitVariation: '',
      variation: true,
      isBackButton: true,
      variationStep: SEQUENTIAL_FORM_FIVE
    };
    const wrapper = shallowWrapperContactDetails(props);
    wrapper.instance().goBack();
    expect(mockCallNextSlide).toHaveBeenCalledWith(expected_input);
  });
});
