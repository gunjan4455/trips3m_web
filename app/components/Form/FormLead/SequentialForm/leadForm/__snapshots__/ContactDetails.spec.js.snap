// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lead form module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactDetails
    callNextSlide={[MockFunction]}
    dispatchRxFormAction={[MockFunction]}
    email={null}
    formValues={
      Object {
        "adult": "2",
        "budget": "5500..6000",
        "budget_new": true,
        "child": "0",
        "child_age_answers_attributes": Array [
          Object {
            "customization_id": "7",
            "option_id": "29",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "30",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "31",
            "user_input": "0",
          },
        ],
        "dont_utilize_budget_new": true,
        "email": null,
        "hotel_category_answers_attributes": Array [
          Object {
            "customization_id": "2",
            "option_id": "7",
          },
        ],
        "hotel_needed_answer_attributes": Object {
          "customization_id": "22",
          "option_id": "95",
        },
        "infant": "0",
        "isd_code": "+91",
        "start_date": "2019-07-25",
        "start_date_flexible": false,
        "start_date_noidea": false,
        "syncErrors": Object {
          "email": "Enter valid email id",
          "phone_no": "Phone number is required",
        },
        "to_loc": "Kerala",
        "trip_days": "2",
      }
    }
    location={
      Object {
        "query": Object {
          "variation": "3",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="contact-icon-container"
        >
          <IconSvg />
        </div>,
        <div
          className="p15"
        >
          <div
            className="pb8"
          >
            <Field
              changeHandler={[Function]}
              component={[Function]}
              emailVal=""
              label="Email ID"
              name="email"
              placeholder="Email"
              type="email"
              validate={
                Array [
                  [Function],
                  [Function],
                ]
              }
            />
          </div>
          <Field
            component={[Function]}
            defaultValue="+91"
            isFocus={false}
            label="Phone No"
            name="phone_no"
            placeholder="Phone"
            type="number"
            validate={[Function]}
          />
          <Field
            className="checkbox-common sfcw"
            classname="at_whatsapp_opt_status"
            component={[Function]}
            formValues={
              Object {
                "adult": "2",
                "budget": "5500..6000",
                "budget_new": true,
                "child": "0",
                "child_age_answers_attributes": Array [
                  Object {
                    "customization_id": "7",
                    "option_id": "29",
                    "user_input": "0",
                  },
                  Object {
                    "customization_id": "7",
                    "option_id": "30",
                    "user_input": "0",
                  },
                  Object {
                    "customization_id": "7",
                    "option_id": "31",
                    "user_input": "0",
                  },
                ],
                "dont_utilize_budget_new": true,
                "email": null,
                "hotel_category_answers_attributes": Array [
                  Object {
                    "customization_id": "2",
                    "option_id": "7",
                  },
                ],
                "hotel_needed_answer_attributes": Object {
                  "customization_id": "22",
                  "option_id": "95",
                },
                "infant": "0",
                "isd_code": "+91",
                "start_date": "2019-07-25",
                "start_date_flexible": false,
                "start_date_noidea": false,
                "syncErrors": Object {
                  "email": "Enter valid email id",
                  "phone_no": "Phone number is required",
                },
                "to_loc": "Kerala",
                "trip_days": "2",
              }
            }
            id="whatsapp_opt_status"
            label="Send trip updates on WhatsApp"
            name="whatsapp_opt_status"
            type="checkbox"
          />
        </div>,
        <Footer
          goBack={[Function]}
          goBackDisabled={false}
          goForward={[Function]}
          goNextDisabled={true}
          isLastSlide={false}
        />,
      ],
      "className": "form-body pt15",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "contact-icon-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="pb8"
            >
              <Field
                changeHandler={[Function]}
                component={[Function]}
                emailVal=""
                label="Email ID"
                name="email"
                placeholder="Email"
                type="email"
                validate={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
              />
            </div>,
            <Field
              component={[Function]}
              defaultValue="+91"
              isFocus={false}
              label="Phone No"
              name="phone_no"
              placeholder="Phone"
              type="number"
              validate={[Function]}
            />,
            <Field
              className="checkbox-common sfcw"
              classname="at_whatsapp_opt_status"
              component={[Function]}
              formValues={
                Object {
                  "adult": "2",
                  "budget": "5500..6000",
                  "budget_new": true,
                  "child": "0",
                  "child_age_answers_attributes": Array [
                    Object {
                      "customization_id": "7",
                      "option_id": "29",
                      "user_input": "0",
                    },
                    Object {
                      "customization_id": "7",
                      "option_id": "30",
                      "user_input": "0",
                    },
                    Object {
                      "customization_id": "7",
                      "option_id": "31",
                      "user_input": "0",
                    },
                  ],
                  "dont_utilize_budget_new": true,
                  "email": null,
                  "hotel_category_answers_attributes": Array [
                    Object {
                      "customization_id": "2",
                      "option_id": "7",
                    },
                  ],
                  "hotel_needed_answer_attributes": Object {
                    "customization_id": "22",
                    "option_id": "95",
                  },
                  "infant": "0",
                  "isd_code": "+91",
                  "start_date": "2019-07-25",
                  "start_date_flexible": false,
                  "start_date_noidea": false,
                  "syncErrors": Object {
                    "email": "Enter valid email id",
                    "phone_no": "Phone number is required",
                  },
                  "to_loc": "Kerala",
                  "trip_days": "2",
                }
              }
              id="whatsapp_opt_status"
              label="Send trip updates on WhatsApp"
              name="whatsapp_opt_status"
              type="checkbox"
            />,
          ],
          "className": "p15",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Field
                changeHandler={[Function]}
                component={[Function]}
                emailVal=""
                label="Email ID"
                name="email"
                placeholder="Email"
                type="email"
                validate={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
              />,
              "className": "pb8",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changeHandler": [Function],
                "component": [Function],
                "emailVal": "",
                "label": "Email ID",
                "name": "email",
                "placeholder": "Email",
                "type": "email",
                "validate": Array [
                  [Function],
                  [Function],
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "defaultValue": "+91",
              "isFocus": false,
              "label": "Phone No",
              "name": "phone_no",
              "placeholder": "Phone",
              "type": "number",
              "validate": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "checkbox-common sfcw",
              "classname": "at_whatsapp_opt_status",
              "component": [Function],
              "formValues": Object {
                "adult": "2",
                "budget": "5500..6000",
                "budget_new": true,
                "child": "0",
                "child_age_answers_attributes": Array [
                  Object {
                    "customization_id": "7",
                    "option_id": "29",
                    "user_input": "0",
                  },
                  Object {
                    "customization_id": "7",
                    "option_id": "30",
                    "user_input": "0",
                  },
                  Object {
                    "customization_id": "7",
                    "option_id": "31",
                    "user_input": "0",
                  },
                ],
                "dont_utilize_budget_new": true,
                "email": null,
                "hotel_category_answers_attributes": Array [
                  Object {
                    "customization_id": "2",
                    "option_id": "7",
                  },
                ],
                "hotel_needed_answer_attributes": Object {
                  "customization_id": "22",
                  "option_id": "95",
                },
                "infant": "0",
                "isd_code": "+91",
                "start_date": "2019-07-25",
                "start_date_flexible": false,
                "start_date_noidea": false,
                "syncErrors": Object {
                  "email": "Enter valid email id",
                  "phone_no": "Phone number is required",
                },
                "to_loc": "Kerala",
                "trip_days": "2",
              },
              "id": "whatsapp_opt_status",
              "label": "Send trip updates on WhatsApp",
              "name": "whatsapp_opt_status",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "goBack": [Function],
          "goBackDisabled": false,
          "goForward": [Function],
          "goNextDisabled": true,
          "isLastSlide": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="contact-icon-container"
          >
            <IconSvg />
          </div>,
          <div
            className="p15"
          >
            <div
              className="pb8"
            >
              <Field
                changeHandler={[Function]}
                component={[Function]}
                emailVal=""
                label="Email ID"
                name="email"
                placeholder="Email"
                type="email"
                validate={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
              />
            </div>
            <Field
              component={[Function]}
              defaultValue="+91"
              isFocus={false}
              label="Phone No"
              name="phone_no"
              placeholder="Phone"
              type="number"
              validate={[Function]}
            />
            <Field
              className="checkbox-common sfcw"
              classname="at_whatsapp_opt_status"
              component={[Function]}
              formValues={
                Object {
                  "adult": "2",
                  "budget": "5500..6000",
                  "budget_new": true,
                  "child": "0",
                  "child_age_answers_attributes": Array [
                    Object {
                      "customization_id": "7",
                      "option_id": "29",
                      "user_input": "0",
                    },
                    Object {
                      "customization_id": "7",
                      "option_id": "30",
                      "user_input": "0",
                    },
                    Object {
                      "customization_id": "7",
                      "option_id": "31",
                      "user_input": "0",
                    },
                  ],
                  "dont_utilize_budget_new": true,
                  "email": null,
                  "hotel_category_answers_attributes": Array [
                    Object {
                      "customization_id": "2",
                      "option_id": "7",
                    },
                  ],
                  "hotel_needed_answer_attributes": Object {
                    "customization_id": "22",
                    "option_id": "95",
                  },
                  "infant": "0",
                  "isd_code": "+91",
                  "start_date": "2019-07-25",
                  "start_date_flexible": false,
                  "start_date_noidea": false,
                  "syncErrors": Object {
                    "email": "Enter valid email id",
                    "phone_no": "Phone number is required",
                  },
                  "to_loc": "Kerala",
                  "trip_days": "2",
                }
              }
              id="whatsapp_opt_status"
              label="Send trip updates on WhatsApp"
              name="whatsapp_opt_status"
              type="checkbox"
            />
          </div>,
          <Footer
            goBack={[Function]}
            goBackDisabled={false}
            goForward={[Function]}
            goNextDisabled={true}
            isLastSlide={false}
          />,
        ],
        "className": "form-body pt15",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "contact-icon-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="pb8"
              >
                <Field
                  changeHandler={[Function]}
                  component={[Function]}
                  emailVal=""
                  label="Email ID"
                  name="email"
                  placeholder="Email"
                  type="email"
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </div>,
              <Field
                component={[Function]}
                defaultValue="+91"
                isFocus={false}
                label="Phone No"
                name="phone_no"
                placeholder="Phone"
                type="number"
                validate={[Function]}
              />,
              <Field
                className="checkbox-common sfcw"
                classname="at_whatsapp_opt_status"
                component={[Function]}
                formValues={
                  Object {
                    "adult": "2",
                    "budget": "5500..6000",
                    "budget_new": true,
                    "child": "0",
                    "child_age_answers_attributes": Array [
                      Object {
                        "customization_id": "7",
                        "option_id": "29",
                        "user_input": "0",
                      },
                      Object {
                        "customization_id": "7",
                        "option_id": "30",
                        "user_input": "0",
                      },
                      Object {
                        "customization_id": "7",
                        "option_id": "31",
                        "user_input": "0",
                      },
                    ],
                    "dont_utilize_budget_new": true,
                    "email": null,
                    "hotel_category_answers_attributes": Array [
                      Object {
                        "customization_id": "2",
                        "option_id": "7",
                      },
                    ],
                    "hotel_needed_answer_attributes": Object {
                      "customization_id": "22",
                      "option_id": "95",
                    },
                    "infant": "0",
                    "isd_code": "+91",
                    "start_date": "2019-07-25",
                    "start_date_flexible": false,
                    "start_date_noidea": false,
                    "syncErrors": Object {
                      "email": "Enter valid email id",
                      "phone_no": "Phone number is required",
                    },
                    "to_loc": "Kerala",
                    "trip_days": "2",
                  }
                }
                id="whatsapp_opt_status"
                label="Send trip updates on WhatsApp"
                name="whatsapp_opt_status"
                type="checkbox"
              />,
            ],
            "className": "p15",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Field
                  changeHandler={[Function]}
                  component={[Function]}
                  emailVal=""
                  label="Email ID"
                  name="email"
                  placeholder="Email"
                  type="email"
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />,
                "className": "pb8",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "changeHandler": [Function],
                  "component": [Function],
                  "emailVal": "",
                  "label": "Email ID",
                  "name": "email",
                  "placeholder": "Email",
                  "type": "email",
                  "validate": Array [
                    [Function],
                    [Function],
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "defaultValue": "+91",
                "isFocus": false,
                "label": "Phone No",
                "name": "phone_no",
                "placeholder": "Phone",
                "type": "number",
                "validate": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "checkbox-common sfcw",
                "classname": "at_whatsapp_opt_status",
                "component": [Function],
                "formValues": Object {
                  "adult": "2",
                  "budget": "5500..6000",
                  "budget_new": true,
                  "child": "0",
                  "child_age_answers_attributes": Array [
                    Object {
                      "customization_id": "7",
                      "option_id": "29",
                      "user_input": "0",
                    },
                    Object {
                      "customization_id": "7",
                      "option_id": "30",
                      "user_input": "0",
                    },
                    Object {
                      "customization_id": "7",
                      "option_id": "31",
                      "user_input": "0",
                    },
                  ],
                  "dont_utilize_budget_new": true,
                  "email": null,
                  "hotel_category_answers_attributes": Array [
                    Object {
                      "customization_id": "2",
                      "option_id": "7",
                    },
                  ],
                  "hotel_needed_answer_attributes": Object {
                    "customization_id": "22",
                    "option_id": "95",
                  },
                  "infant": "0",
                  "isd_code": "+91",
                  "start_date": "2019-07-25",
                  "start_date_flexible": false,
                  "start_date_noidea": false,
                  "syncErrors": Object {
                    "email": "Enter valid email id",
                    "phone_no": "Phone number is required",
                  },
                  "to_loc": "Kerala",
                  "trip_days": "2",
                },
                "id": "whatsapp_opt_status",
                "label": "Send trip updates on WhatsApp",
                "name": "whatsapp_opt_status",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "goBack": [Function],
            "goBackDisabled": false,
            "goForward": [Function],
            "goNextDisabled": true,
            "isLastSlide": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
